src/classes/PMT_Payment_TEST.cls

```diff
@@ -56,44 +56,124 @@ private class PMT_Payment_TEST {
+     * @description Validate that new payment records created with an opportunity get the CurrencyIsoCode set
+     * properly to the Opportunities currency code
      */
-    static testMethod void testPaymentCurrIsInSyncWithOppCurrInMultiCurrencyOrg() {
+    static testMethod void test_newPaymentCurrencyMatchesOpportunity() {
 
         //only run test in multicurrency orgs
         if (!userInfo.isMultiCurrencyOrganization()) {
             return;
         }
 
-        setupISOCodes();
-
-        UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_And_Orgs_Settings__c (npe01__Payments_Enabled__c= true));
+        Account acc = [SELECT Id FROM Account LIMIT 1];
 
-        Account acc = new Account(Name='foo');
-        insert acc;
+        setupISOCodes();
+        // If there is no active non-default currency just exit out at this point.
+        if (nonDefaultCurrencyType == null) {
+            return;
+        }
 
         Test.startTest();
 
         Opportunity opp = new Opportunity(
-                Name='success',
+                Name = 'success',
                 Amount = 10,
-                AccountID=acc.id,
-                CloseDate=system.today(),
-                StageName=UTIL_UnitTestData_TEST.getClosedWonStage());
+                AccountID = acc.id,
+                CloseDate = system.today(),
+                StageName = UTIL_UnitTestData_TEST.getClosedWonStage());
+        opp.put('CurrencyIsoCode', (String) nonDefaultCurrencyType.get('IsoCode'));
         insert opp;
+
+        Test.StopTest();
+
         String soql = 'SELECT Id, CurrencyIsoCode FROM Opportunity WHERE Id = \'' + opp.Id + '\' LIMIT 1';
         opp = database.query(soql);
 
         npe01__OppPayment__c pmt = getPaymentRecord(opp.Id);
-        System.AssertEquals((String)opp.get('CurrencyIsoCode'), (String)pmt.get('CurrencyIsoCode'));
+        System.AssertEquals((String) opp.get('CurrencyIsoCode'), (String) pmt.get('CurrencyIsoCode'));
+    }
+
+    /**
+     * @description Validate that a change to the CurrencyIsoCode on the Opportunity record is copied
+     * down the Payment records.
+     */
+    static testMethod void test_UpdatePaymentCurrencyFromOpportunity() {
+
+        //only run test in multicurrency orgs
+        if (!userInfo.isMultiCurrencyOrganization()) {
+            return;
+        }
 
+        setupISOCodes();
         // If there is no active non-default currency just exit out at this point.
         if (nonDefaultCurrencyType == null) {
             return;
         }
 
+        Account acc = [SELECT Id FROM Account LIMIT 1];
+
+        Test.startTest();
+
+        Opportunity opp = new Opportunity(
+                Name = 'success',
+                Amount = 10,
+                AccountID = acc.id,
+                CloseDate = system.today(),
+                StageName = UTIL_UnitTestData_TEST.getClosedWonStage());
+        insert opp;
+
+        npe01__OppPayment__c pmt = getPaymentRecord(opp.Id);
```

Comment:
Do we need to verify the Payment currency after retrieving the Payment on this line?

Reply:
No. That functionality is verified in the previous test.

Reply:
So we do not retrieve the Payment here if we are not going to check it or use it before line 142?

Reply:
Yeah, true. I left the the retrieve up there when technically it's not needed until a few lines down. I will fix.